### DISCUSSION TITLE
Basic Question - How to add plugins in data-prepper

### DISCUSSION THREAD

**Discussion by srinioffice** [opensearch-project/data-prepper#2915] [Q&A]
I have installed data-prepper from tar (not docker). I want to add plugins like rss to my data-prepper. Can someone help me the procedure. Here are some of the basic questions I have.

1. Under which directory I should place the plugins?
2. From where I should download the plugins?
3. Are there any installation and configuration guide for rss-source plugin?

Any help is much appreciated.

**Comment by dlvenable** [comment]
@srinioffice , Thank you for trying out Data Prepper.

Currently, all of the Data Prepper plugins are pre-packaged and installed with Data Prepper - both for the Docker image and the tar.gz. So you should get them out of the box.

**Reply by umesh1103-asus** [reply]
So @dlvenable isn’t there any way to add our own custom processors ??

**Reply by dlvenable** [reply]
@umesh1103-asus ,

It is possible to add your own custom processors. Is this something you'd like to contribute back to the project or is it specific to your situation? If it is somewhat generic, we would welcome a PR for it. 

You can see this project that shows how to setup a processor for Data Prepper without adding it back to the project.

https://github.com/dlvenable/data-prepper-sample-plugin

You will need to package them in a custom tar.gz or Docker image. We have an open issue to support downloading plugins from Maven jars (#321). But, this is not yet implemented.
